akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -974,3 +975,38 @@ private[stream] class DropWithin[T](timeout: FiniteDuration) extends SimpleLinea
 
   override def toString = "DropWithin"
 }
+
+/**
+ * INTERNAL API
+ */
+private[stream] final class Reduce[T](f: (T, T) ⇒ T) extends SimpleLinearGraphStage[T] {
+  override def initialAttributes: Attributes = DefaultAttributes.reduce
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
+    override def toString = s"Reduce.Logic(aggregator=$aggregator)"
+    var aggregator: T = _
+
+    setHandler(in, new InHandler {
+      override def onPush(): Unit = {
+        aggregator = grab(in)
+        pull(in)
+        setHandler(in, rest)
+      }
+    })
+    val rest = new InHandler {
```

Comment:
not really. In this case it being a val means an extra reference stored although this instance is not referenced anymore by the stage itself, so it can be a def. Not much of an issue though, but since it is cheap to fix let's make it a def.